### PR TITLE
[IMP] im_livechat, *: introduce PublicLivechatGlobal/loadQWebTemplate

### PR DESCRIPTION
**Pull request by alexkuhn**
*: website_livechat

Task-2937799

### CODE REVIEW

File: addons/im_livechat/static/src/models/public_livechat_global.js [hunk 1]
```diff
@@ -28,6 +29,12 @@ registerModel({
         },
     },
     recordMethods: {
+        async loadQWebTemplate() {
+            const templates = await this.messaging.rpc({ route: '/im_livechat/load_templates' });
+            for (let template of templates) {
+                qweb.add_template(template);
```
Comment (oomsveta) [inline addons/im_livechat/static/src/models/public_livechat_global.js:35]:
```suggestion
                QWeb.add_template(template);
```

File: addons/im_livechat/static/src/models/public_livechat_global.js [hunk 2]
```diff
@@ -28,6 +29,12 @@ registerModel({
         },
     },
     recordMethods: {
+        async loadQWebTemplate() {
+            const templates = await this.messaging.rpc({ route: '/im_livechat/load_templates' });
+            for (let template of templates) {
```
Comment (oomsveta) [inline addons/im_livechat/static/src/models/public_livechat_global.js:34]: while you are at it:
```suggestion
            for (const template of templates) {
```

**Review by oomsveta — CHANGES_REQUESTED** [review]

File: addons/im_livechat/static/src/models/public_livechat_global.js (hunk 1 above)
Comment (alexkuhn) [reply addons/im_livechat/static/src/models/public_livechat_global.js:35]: No, it follows the import, `qweb !== QWeb`. Previous code introduced an unnecessary local variable `QWeb`, so I removed it and just used the actual named export stuff

File: addons/im_livechat/static/src/models/public_livechat_global.js (hunk 2 above)
Comment (alexkuhn) [reply addons/im_livechat/static/src/models/public_livechat_global.js]: change done

**Comment by alexkuhn** [comment]
Merged in grouped PR: https://github.com/odoo/odoo/pull/97219

**Review by oomsveta — APPROVED** [review]

**Pull request closed without merging**